application/models/User.php

```diff
@@ -972,15 +967,12 @@ public function getGroupMemberListButtons()
     /**
      * Returns true if logged in user with id $loginId can edit this user
      *
-     * @param int $loginId
      * @return bool
      */
-    public function canEdit($loginId)
+    public function canEdit()
```

Comment:
Changing the signature of the public function?
If so, why not just remove it? Seems to be just an alias.

Reply:
1. There are 2 function to do the same things …
2. You can not get App()->getUser() with `$loginId`
3. You right, maybe remove 

Or create a 3rd one … with both are shortcut for this 3rd one …  and sometime someone look at other function by other dev and not create new functions again and again …


Then what do you want

1. Stay like that
2. remove this one
3. Create a third one and the 2 other call the 3rd one

Reply:
Let's resolve the other comments (https://github.com/LimeSurvey/LimeSurvey/pull/3495#discussion_r1338619193) first

Reply:
> Let's resolve the other comments ([#3495 (comment)](https://github.com/LimeSurvey/LimeSurvey/pull/3495#discussion_r1338619193)) first

In my opinion : there are nothing to resolve here … if we stay like 3,5 and 6 : we need to disallow user to update himself on this way.

And this don't touch this part about canEdit function …

Reply:
I would just leave this function after all

Reply:
> I would just leave this function after all

After reflection : i think you're right here : set it s deprecated (both) and use a new one ?

User->hasPermission ? update and create are specific

Reply:
> I would just leave this function after all

Other solution : move function logic here, Service function are shortcut.

Reply:
> Changing the signature of the public function?

No change now :)